Core/clim-core/commands.lisp

```diff
@@ -1294,33 +1294,54 @@ examine the type of the command menu item to see if it is
 (define-presentation-method presentation-typep (object (type command))
   (and (consp object)
        (presentation-typep (car object)
-			   `(command-name :command-table ,command-table))))
+                           `(command-name :command-table ,command-table))))
 
 (define-presentation-method presentation-subtypep ((type command)
-						   maybe-supertype)
+                                                   maybe-supertype)
   (with-presentation-type-parameters (command maybe-supertype)
     (let ((super-table command-table))
       (with-presentation-type-parameters (command type)
-	(command-table-inherits-from-p super-table command-table)))))
+        (command-table-inherits-from-p super-table command-table)))))
 
 (define-presentation-method present (object (type command)
-				     stream
-				     (view textual-view)
-				     &key acceptably for-context-type)
+                                     stream
+                                     (view textual-view)
+                                     &key acceptably for-context-type)
   (declare (ignore acceptably for-context-type))
   (funcall *command-unparser* command-table stream object))
 
 (define-presentation-method accept ((type command) stream
-				    (view textual-view)
-				    &key)
-  (let ((command (funcall *command-parser* command-table stream)))
-    (cond ((null command)
-	   (simple-parse-error "Empty command"))
-          ((partial-command-p command)
-           (funcall *partial-command-parser*
-            command-table stream command
-            (position *unsupplied-argument-marker* command)))
-	  (t (values command type)))))
+                                    (view textual-view)
+                                    &key)
+  (setq command-table (find-command-table command-table))
+  (let ((start-position (and (input-editing-stream-p stream)
+                             (stream-scan-pointer stream)))
+        (replace-input-p nil))
+    (multiple-value-bind (object new-type)
+        ;; We establish a new input context so that clicks on command-menu-items
+        ;; throw to us. This will let's us handle "partial commands" below.
+        ;; This only works for partial commands coming by clicking on
+        ;; application-frame menus.  Clicking on a presentation of something as a
+        ;; command with unsupplied arguments with instead by caught by accept-1
+        ;; and bypass this handling.
+        (with-input-context ('command :override nil)
+            (object presentation-type event options)
+            (funcall *command-parser* command-table stream)
```

Comment:
this should be `(values (funcall …) type)`, otherwise sanitizer will receive ptype null